Project proposal applying to move Aries Agent Test Harness to OWF as the OWL project

swcurran opened the pull request:
A project proposal applying to move a component of Hyperledger Aries -- Aries Agent Test Harness -- to OWF.

Agent Test Harness is an extensible open-source testing framework designed to validate the functionality and interoperability of agents and components within the decentralized trust ecosystem. A summary of its daily interoperability test runs are published on the website [https://aries-interop.info](https://aries-interop.info). The test harness is crucial for ensuring that different decentralized trust agent implementations, which are responsible for handling decentralized identifiers (DIDs) and exchanging verifiable credentials, can interoperate seamlessly. Agent Test Harness currently includes support for the OWF’s Credo-TS project, and is used by the OWF’s Mobile Agent Test Harness project. The proposal also includes moving into the same project [Aries Akrida](https://github.com/hyperledger/aries-akrida), a scriptable tool for generating loads to apply to Aries (and potential, other) wallets/agents.

@nodlesh, @TelegramSam, @tkuhrt, @KimEbert42 @TimoGlastra @ianco

This PR is associated with https://github.com/openwallet-foundation/project-proposals/pull/42.

KimEbert42 commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,101 @@
+| --------------- | ----------- | ---------------------------------------- |
+| Sheldon Regular | nodlesh     | Funded by Government of British Columbia |
+| Stephen Curran  | swcurran    | Funded by Government of British Columbia |
+| Kim Ebert       | KimEbert42  | Indicio, PBC                             |
```
I believe we should include Alex Walker | anwalker293 | Inidicio, PBC in the list of maintainers.

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+
+## Preferred Maturity Level
+
+We propose that the project be at the “Growth” level.  All three of the components have been used extensively in the past and are well documented. Many groups have used the tools in testing their own instances of the various support agents and wallets.  As we move the project to OWF and start up the Wallet Interoperability Special Interest Group (link to PR to be added) these tools will be useful ...
```
Can you comment on how the project meets each of the "Growth" acceptance criteria outlined in the [project lifecycle](https://tac.openwallet.foundation/governance/project-lifecycle/)?

swcurran replied:
Added how the Growth stage requirements are met to the proposal.

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+* Automation: The Agent Test Harness automates the testing of agents, making it easier for developers to continuously verify the compliance and performance of their implementations. Developers can configure what agent implementations with whom they want to test interoperability and for each, what specific tests they want executed. Implementations can even include their CI pipelines select Agent T ...
+* Extensibility: The framework is highly extensible, allowing developers to add their implementations for testing and to construct custom tests and scenarios that suit a wide variety of use cases. This flexibility is important for testing a wide range of decentralized identity solutions across different industries.
+* Comprehensive Coverage: The test harness covers a wide array of functionalities that decentralized trust agents should support, including secure messaging, protocols, and credential issuance, verification and revocation across a range of credential formats.
+* Community-Driven: Agent Test Harness is intended to be developed and maintained by a global community of contributors to ensure that it stays up-to-date with the latest standards and best practices in the decentralized identity space.
```
It would be interesting to see what might be done with some of the labs that we have that would allow us to add sd-jwt interoperability testing to this project (if it does not already exist).

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+* Comprehensive Coverage: The test harness covers a wide array of functionalities that decentralized trust agents should support, including secure messaging, protocols, and credential issuance, verification and revocation across a range of credential formats.
+* Community-Driven: Agent Test Harness is intended to be developed and maintained by a global community of contributors to ensure that it stays up-to-date with the latest standards and best practices in the decentralized identity space.
+
+As mentioned in the [Project Name](#project-name-to-be-determined) section, we suggest that Agent Test Harness be combined into a single project with the Mobile Wallet Test Harness and Akrida load testing tool. All will be separate repos in the project, but share common project components (governance, meetings, Discord community, etc.).
```
Ideally for consistency with previous OWF decisions, we could then make each repo have the same project prefix (e.g., project-name-repo-name).

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+
+1. Interoperability: The primary purpose of Agent Test Harness is to ensure interoperability between different agents by validating their ability to communicate using standard protocols like DIDComm and to handle core functions like verifiable credential exchanges. It provides a consistent testing environment to verify that agents from different developers and organizations, evolving independentl ...
+2. Promotion of Open Standards: As an open-source tool, Agent Test Harness is built around open standards and protocols, ensuring that the agents it tests comply with widely accepted norms in the decentralized identity space. This approach encourages the adoption of these standards across the industry, fostering a more unified and interoperable ecosystem.
+3. Facilitating Innovation and Adoption:  By providing developers with a tool to validate their agents with those developed by others in the space, Agent Test harness lowers the barriers to entry for creating new, compliant, and interoperable decentralized identity solutions. This facilitates innovation and encourages the adoption of decentralized identity technologies.
```
```suggestion
3. Facilitating Innovation and Adoption:  By providing developers with a tool to validate their agents with those developed by others in the space, Agent Test Harness lowers the barriers to entry for creating new, compliant, and interoperable decentralized identity solutions. This facilitates innovation and encourages the adoption of decentralized identity technologies.
```

swcurran replied:
Fixed

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+
+## Code of Conduct
+
+ACA-Py follows the [Hyperledger Code of Conduct](https://github.com/hyperledger/aries-agent-test-harness/blob/main/CODE_OF_CONDUCT.md), which is what the OWF code of conduct is based on.
```
```suggestion
Agent Test Harness follows the [Hyperledger Code of Conduct](https://github.com/hyperledger/aries-agent-test-harness/blob/main/CODE_OF_CONDUCT.md), which is what the OWF code of conduct is based on.
```

swcurran replied:
Fixed

tkuhrt commented on projects/agent-test-harness.md:
```diff
@@ -0,0 +1,102 @@
+
+## Proposed Project Governance
+
+The current governance model under Hyperledger is consensus-based. This means that decisions are made through discussions, with the aim of community consensus, as outlined in the [Aries Project Charter](https://docs.google.com/document/d/1F6RbR7xDaBt5CDJhqLJzR4c1pDJtyPGshp9fy6eVtSM/edit?usp=sharing). In cases where no clear consensus is established, a project Technical Steering Committee, or the  ...
```
Is the intention for this project to be distinct from the other Aries proposals and their governance processes when it moves to OWF?

swcurran replied:
Agent Test is proposed to be a separate project at OWF, independent of the other artifacts stemming from the existing Hyperledger Aries project, with separate governance. That reflects the distinct purpose of the project (and its sub-projects), and community around it.

tkuhrt reviewed (changes requested):
Thanks for the proposal, @swcurran. I have added this proposal to Wednesday's TAC meeting (September 18th). We will cover this proposal after the Askar proposal (assuming time remaining).

darrellodonnell commented:
We support this project at GAN and will be actively contributing shortly.

swcurran commented:
Updated the project name and the project proposal file name.  Adjusted the wording when referring to the overall project (OWL) and the entire toolset vs. sections talking about the individual tools -- Agent Test Harness, Akrida, and Mobile Wallet Test Harness.

@nodlesh
